### PR TITLE
doc: release notes: Add Zephyr 2.2 release notes for Bluetooth

### PR DESCRIPTION
**Pull request by jhedberg**
Add release notes for Bluetooth.

### CODE REVIEW

**Comment by jhedberg** [comment]
@carlescufi @cvinayak could you add patches with the controller changes to this PR?

File: doc/releases/release-notes-2.2.rst [hunk 1]
```diff
@@ -429,7 +430,21 @@ Bluetooth
 
 * Host:
 
-  * <TBD>
+  * GAP: support for Extended Advertising
```
Comment (joerchan) [inline doc/releases/release-notes-2.2.rst]: We haven't added this. Is there a milestone somewhere that needs to be updated?

File: doc/releases/release-notes-2.2.rst [hunk 2]
```diff
@@ -429,7 +430,21 @@ Bluetooth
...
+  * GAP: Add dynamic LE scan listener API
+  * GAP: Pre-allocate connection objects for connectable advertising
```
Comment (joerchan) [inline doc/releases/release-notes-2.2.rst]: As well as whitelist initiatior (`bt_conn_create_le_auto`)

File: doc/releases/release-notes-2.2.rst (hunk 1 above)
Comment (carlescufi) [inline doc/releases/release-notes-2.2.rst]: This is not merged, is it?

File: doc/releases/release-notes-2.2.rst [hunk 3]
```diff
@@ -429,7 +430,21 @@ Bluetooth
...
+  * GAP: Fixes for multi-identity support
+  * GAP: RPA timeout handling fixes
+  * L2CAP: move channel processing to the system workqueue
+  * L2CAP: multiple fixes for credit-based flowcontrol
+  * SMP: Fix Security Manager timeout handling
```
Comment (joerchan) [inline doc/releases/release-notes-2.2.rst]: I think we could add the following:
SMP: Add pairing_accept callback.
GATT: Add return value to cfg_write callback.
GAP: Add remote version information.

File: doc/releases/release-notes-2.2.rst (hunk 1 above)
Comment (jhedberg) [reply doc/releases/release-notes-2.2.rst]: No, my mistake. Will remove.

Comment (jhedberg) [reply doc/releases/release-notes-2.2.rst]: Removed now

Comment (jhedberg) [reply doc/releases/release-notes-2.2.rst]: Already removed, sorry

File: doc/releases/release-notes-2.2.rst (hunk 2 above)
Comment (jhedberg) [reply doc/releases/release-notes-2.2.rst]: Will add

File: doc/releases/release-notes-2.2.rst (hunk 3 above)
Comment (jhedberg) [reply doc/releases/release-notes-2.2.rst]: Added now

**Review by trond-snekvik — APPROVED** [review]

**Review by joerchan — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**